src/MessagingAttachment/MessagingAttachment.tsx

```diff
@@ -12,27 +13,41 @@ function cssClass(element: string) {
 
 type Props = {
   attachmentID: string;
+  // TODO: should this be optional?
+  attachmentURL: string;
   errorMsg?: string;
-  icon: React.ReactNode;
+  // TODO: is this needed, or only mimeType?
```

Comment:
Seems like only one is required, so whichever is more convenient? We should always have both available from the consumer side.

Reply:
Would it be unreasonable to use the mime package to get the mimetype from the filetype? if its reasonable, then I can remove the mimetype as a prop

Reply:
I think making the consumer pass it in is better, so that the component doesn't have to do any conversions itself. No harm in keeping it as a prop!

Reply:
Actually, its a little hacky, but I can use `fileTypeToIcon()` ([link](https://github.com/Clever/components/blob/master/src/MessagingAttachment/MessagingAttachment.tsx#L99)) to achieve this